reconciletags/reconciletags_e2e_test.py

```diff
@@ -0,0 +1,124 @@
+                                     '--format=json', repo]))
+        return output
+
+    def _BuildImage(self, full_image_name):
+        # create a non-functional but tiny docker image
+        subprocess.call(['gcloud', 'beta', 'container', 'builds',
+                         'submit', self._DIR, '-q', '--tag', full_image_name])
+
+        # create the json config file and write to it
+        self.tmpdir = tempfile.mkdtemp()
+        self.full_filename = os.path.join(self.tmpdir, self._FILE_NAME)
+        test_json = open(self.full_filename, 'w')
+
+        # grab the just created digest
+        output = self._ListTags(self._REPO)
+        self.assertEqual(len(output), 1)
+        self.digest = output.pop()['digest'].split(':')[1]
+
+        # write the proper json to the config file
+        self._TEST_JSON['projects'][0]['images'][0]['digest'] = self.digest
+        json.dump(self._TEST_JSON, test_json)
+
+    def setUp(self):
+        self._BuildImage(self._REPO + ':' + self._TAG)
+
+    def tearDown(self):
+        subprocess.call(['gcloud', 'beta', 'container', 'images',
+                         'delete', self._REPO + '@sha256:' + self.digest,
+                         '-q'])
+        shutil.rmtree(self.tmpdir)
+
+    def testTagReconciler(self):
+        # run the reconciler
+        subprocess.check_output(['python', 'reconciletags.py',
+                                 self.full_filename])
+
+        # check list-tags to see if it added the correct tag
+        output = self._ListTags(self._REPO)
+        for image in output:
+            if image['digest'].split(':')[1] == self.digest:
+                self.assertEquals(len(image['tags']), 2)
+                self.assertEquals(image['tags'][1], 'testing')
+                self.assertEquals(image['tags'][0], self._TAG)
+
+        # run reconciler again and make sure nothing changed
+        subprocess.check_output(['python', 'reconciletags.py',
+                                 self.full_filename])
+
+        output = self._ListTags(self._REPO)
+        for image in output:
+            if image['digest'].split(':')[1] == self.digest:
+                self.assertEquals(len(image['tags']), 2)
+                self.assertEquals(image['tags'][1], 'testing')
+                self.assertEquals(image['tags'][0], self._TAG)
+
+    def testWithFakeDigest(self):
+        # now run it with a fake digest
+        fake_digest_filename = os.path.join(self.tmpdir, 'fake_digest.json')
+        fake_digest_json = open(fake_digest_filename, 'w')
+
+        self._TEST_JSON['projects'][0]['images'][0]['digest'] = 'fakedigest'
+        json.dump(self._TEST_JSON, fake_digest_json)
+        fake_digest_json.close()
+
+        # we need to look at the actual error message instead of
+        # using a generic self.assertRaises since all errors
+        # will raise a CalledProcessError regardless of what failed.
+        p = subprocess.Popen(['python', 'reconciletags.py',
```

Comment:
actually it might make more sense to just call this in process, then you can check the error a little easier. What do you think about importing the code and just calling the main function with the right arguments?